File: ex2.c
#include <stdio.h>
#include <stdlib.h>

int vagas[2][10];

void limparTela(){
    #ifdef _WIN32
    system("cls");   // roda isso no Windows
    #else
    system("clear"); // roda isso no Linux/Mac
    #endif
}

void aguardar() {
    printf("\nPressione ENTER para continuar...");
    while (getchar() != '\n');
    while (getchar() != '\n');
}

void adicionarCarro(){
    int andar;
    int vaga;
    do{
        printf("Qual o andar da vaga (1 = terreo / 0 = subsolo): ");
        scanf("%d", &andar);
        if(andar != 0 && andar != 1) printf("Andar invalido. Tente novamente.\n");
    } while(andar != 0 && andar != 1);

    do{
        printf("Qual o numero da vaga (0 - 9): ");
        scanf("%d", &vaga);
        if(vaga < 0 || vaga > 9) printf("Vaga invalida. Tente novamente.\n");
    } while(vaga < 0 || vaga > 9);

     if (vagas[andar][vaga] == 1) {
        printf("\nEssa vaga ja esta ocupada!\n");
    } else {
        vagas[andar][vaga] = 1;
        printf("\nCarro adicionado com sucessos\n");
    }

    aguardar();
}

void removerCarro(){

    int andar;
    int vaga;
    do{
        printf("Qual o andar da vaga que deseja remover (1 = terreo / 0 = subsolo): ");
        scanf("%d", &andar);
        if(andar != 0 && andar != 1) printf("Andar invalido. Tente novamente.\n");
    } while(andar != 0 && andar != 1);

    do{
        printf("Qual o numero da vaga que deseja remover (0 - 9): ");
        scanf("%d", &vaga);
        if(vaga < 0 || vaga > 9) printf("Vaga invalida. Tente novamente.\n");
    } while(vaga < 0 || vaga > 9);

    if(vagas[andar][vaga] == 0){
        printf("Essa vaga ja esta vazia!\n");
    } else {
        vagas[andar][vaga] = 0;
        printf("\nCarro removido com sucesso!\n");
    }

    aguardar();
}

void vagasPorAndar(){
    int andar;
    int colunas = sizeof(vagas[0]) / sizeof(vagas[0][0]);
    do{

        printf("Qual andar deseja verificar quais vagas estao livres (1 = terreo / 0 = subsolo): ");
        scanf("%d", &andar);
        if(andar != 0 && andar != 1) printf("Andar invalido. Tente novamente.\n");
    } while(andar != 0 && andar != 1);

    int livres = 0;
    for(int i = 0; i < colunas; i++){
        if(vagas[andar][i] == 0){
            printf("Vaga [%d] esta vazia\n", i);
            livres++;
        }
    }

    if (livres == 0) printf("  Nenhuma vaga livre neste andar.\n");
    else printf("\nTotal: %d vaga(s) livre(s).\n", livres);

    aguardar();
}

void vagasEstacionamento(){
    int colunas = sizeof(vagas[0]) / sizeof(vagas[0][0]);

    for(int i = 0; i < colunas; i++){
        if(vagas[1][i] == 0){
            printf("Terreo: vaga [%d] esta disponivel\n", i);
        }
    }

    printf("\n");

    for(int i = 0; i < colunas; i++){
        if(vagas[0][i] == 0){
            printf("Subsolo: vaga [%d] esta disponivel\n", i);
        }
    }

    aguardar();
}

int main(){

    int escolha;


    do {
        limparTela();
        printf("Escolha uma opcao\n");
        printf("\n");
        printf("  [1]  Adicionar carro\n");
        printf("  [2]  Remover carro\n");
        printf("  [3]  Vagas por andar\n");
        printf("  [4]  Vagas no estacionamento\n");
        printf("  [0]  Sair\n");
        printf("\n");
        if(scanf("%d", &escolha) != 1){
            printf("Erro: Digite apenas os numeros indicados.");
            return 0;
        }

        switch (escolha){
            case 1:
            
            adicionarCarro();
            break;
            case 2:
            removerCarro();
            break;
            case 3:
            vagasPorAndar();
            break;
            case 4:
            vagasEstacionamento();
            break;
            case 0:
            printf("Encerrando o sistema. Ate logo!\n\n");
            break;
            default:
            printf("Erro: Opcao invalida.\n\n");
            aguardar();
        }

    } while (escolha != 0);

    return 0;
}

File: ex2.py
import os

vagas = [
    [0,0,0,0,0,0,0,0,0,0], # subsolo (0)
    [0,0,0,0,0,0,0,0,0,0]  # subsolo (1)
]

def limpar_tela():
    if(os.name == 'nt'):    # Windows
        os.system("cls")    
    else:
        os.system("clear")  # Linux / Mac

def aguardar():
    input("Pressione ENTER para continuar...")

def adicionarCarro():
    while True: # Semelhante ao Do-While em C
        andar = int(input("Qual o andar da vaga (1 = terreo / 0 = subsolo): "))
        if andar == 0 or andar == 1:
            break
        print("Erro: Andar invalido. Tente novamente!")
    
    while True:
        vaga = int(input("Qual o numero da vaga (0 - 9): "))
        if vaga >= 0 and vaga <= 9:
            break
        print("Erro: Vaga invalida. Tente novamente!")
    
    if vagas[andar][vaga] == 1:
        print("\nEssa vaga ja esta ocupada!")
    else:
        vagas[andar][vaga] = 1
        print("\nCarro adicionado com sucesso!")

    aguardar()

def removerCarro():
    while True:
        andar = int(input("Qual o andar da vaga que deseja remover (1 = terreo / 0 = subsolo): "))
        if andar == 0 or andar == 1:
            break
        print("Erro: Andar invalido. Tente novamente!")
    
    while True:
        vaga = int(input("Qual o numero da vaga que deseja remover (0 - 9): "))
        if vaga >= 0 and vaga <= 9:
            break
        print("Erro: Vaga invalida. Tente novamente!")

    if vagas[andar][vaga] == 0:
        print("\nEssa vaga ja esta vazia!")
    else:
        vagas[andar][vaga] = 0
        print("\nCarro removido com sucesso!")
    
    aguardar()

def vagasPorAndar():
    colunas = len(vagas[0]) # colunas = 10

    while True:
        andar = int(input("Qual andar deseja verificar quais vagas estao livres (1 = terreo / 0 = subsolo): "))
        if andar == 0 or andar == 1:
            break
        print("Erro: Andar invalido. Tente novamente!")
    
    livres = 0
    for i in range(0, colunas):
        if vagas[andar][i] == 0:
            print(f"Vaga {[i]} esta vazia")
            livres+=1
    
    if livres == 0:
        print("Nenhuma vagas livre neste andar.")
    else:
        print(f"\nTotal: {livres} vaga(s) livres(s).")
    
    aguardar()

def vagasEstacionamento():
    colunas = len(vagas[0])
    total = 0

    livres1 = 0
    print("Terreo:")
    for i in range(0, colunas):
        if vagas[1][i] == 0:
            print(f"  Vaga {[i]} disponivel")
            livres1 += 1

    if livres1 == 0:
        print("Nenhuma vaga livre.\n")
    total += livres1

    print("\nSubsolo:")
    livres0 = 0
    for i in range(0, colunas):
        if vagas[0][i] == 0:
            print(f"  Vaga {[i]} disponivel")
            livres0 += 1
    
    if livres0 == 0:
        print("Nenhuma vaga livre.\n")
    total += livres0

    print(f"\nTotal geral: {total} vaga(s) livre(s).")

    aguardar()

if __name__ == "__main__":

    while True:
        limpar_tela()
        print("Escolha uma opcao\n")
        print("  [1]  Adicionar carro")
        print("  [2]  Remover carro")
        print("  [3]  Vagas por andar")
        print("  [4]  Vagas no estacionamento")
        print("  [0]  Sair\n")

        escolha = int(input())

        match escolha:
            case 1:
                adicionarCarro()
            case 2:
                removerCarro()
            case 3:
                vagasPorAndar()
            case 4:
                vagasEstacionamento()
            case 0:
                print("Encerrando o sistema. Ate logo!\n")
                break
            case _:
                print("Erro: Opcao invalida.\n")
                aguardar()